app/api/search/specs/entitiesIndex.spec.js

```diff
@@ -78,4 +86,80 @@ describe('entitiesIndex', () => {
       ]);
     });
   });
+
+  describe('updateMapping', () => {
+    it('should update the mapping provided by the factory', async () => {
+      const templates = [
+        {
+          _id: '123',
+          name: 'test',
+          properties: [
+            { name: 'name', type: 'text' },
+            { name: 'dob', type: 'date' },
+            { name: 'country', type: 'select' },
+          ],
+        },
+      ];
+      await updateMapping(templates, elasticIndex);
+      const mapping = await elastic.indices.getMapping({ index: elasticIndex });
+      const mappedProps = mapping.body[elasticIndex].mappings.properties.metadata.properties;
+      expect(mappedProps.name).toMatchSnapshot();
+      expect(mappedProps.dob).toMatchSnapshot();
+      expect(mappedProps.country).toMatchSnapshot();
+    });
+  });
+
+  describe('checkMapping', () => {
+    it('should check mapping of a template vs current mapping', async () => {
+      const templateA = {
+        _id: '123',
+        name: 'template A',
+        properties: [
+          { name: 'name', type: 'text', label: 'Name' },
+          { name: 'dob', type: 'date', label: 'Date of birth' },
+          { name: 'country', type: 'select', label: 'Country' },
+        ],
+      };
+
+      const templateB = {
+        _id: '456',
+        name: 'template B',
+        properties: [{ name: 'dob', type: 'date', label: 'Date of birth' }],
+      };
+
+      await updateMapping([templateA], elasticIndex);
+      let response = await checkMapping(templateB, elasticIndex);
+      expect(response).toEqual({ errors: [], valid: true });
+
+      templateB.properties[0].type = 'text';
+      response = await checkMapping(templateB, elasticIndex);
+      expect(response).toEqual({ errors: [{ name: 'Date of birth' }], valid: false });
+    });
+  });
+
+  describe('reindexAll', () => {
+    it('should delete a field from the mapping', async () => {
```

Comment:
This test passes green even if I don't call on `indexEntities` and complete delete line `174` from `entitiesIndex`.

Not the actual return nor the crucial step of indexing all entities is tested at all.